Use colors to flag print state

vo689bj opened the pull request:
Hi,, i separate table printjob by coloring each row depends on state. Please check if it is ok for you.

vo689bj committed:
created accessor getOriginStateAttrbitue on PrintJob model.

vo689bj committed:
added new column origin_state which is hidden. It was needed for identified job state like CANCELLED not Cancelled!!!!!!

kdmnk reviewed (changes requested):
Hi! Thanks!
I like your approach of using accesors!
However, coloring the full row can be too much for the eyes, I think something like colorful badges are what we want here (sorry for the unclear issue). 
One way to achieve this is to use a cell formatter function (like the `dateFormatter` in this table) with a code like this:
`value = '<span class="badge '+colors[cell.getRow().getData().origin_state]+'">'+value+'</span>';`

The problem why your solution does not work properly is that the origin states get translated while displayed in the frontend, so your `row.getData().origin_state == ...` will be false. Currently I don't see what would be the solution here, maybe @luksan47 has an idea.

Lastly, please make sure that the text is always readable in the colorful background.

kdmnk commented on resources/views/dormitory/print/history.blade.php:
```diff
@@ -85,8 +85,32 @@
+                        SUCCESS: "green",
+                    };
+
+                    @foreach(\App\Models\PrintJob::STATES as $key => $state)
+                        var existColor = "{{$state}}" in colors;
+                        if (row.getData().origin_state == "{{$state}}" && existColor) {
+                            const children = row.getElement().childNodes;
+                            children.forEach((child) => {
+                                child.style.backgroundColor = colors["{{$state}}"];
+                            });
+                        }
+                    @endforeach
```
FYI You could write this instead of the whole foreach: 
```
row.getElement().childNodes.forEach((child) => {
                                child.style.backgroundColor = colors[row.getData().origin_state];
                            });
```
;)

luksan47 commented:
Thanks @vo689bj for your PR and sorry for not replying under the issue, it was a busy week for us!

You picked the right colours for the print states, but it would be better with a lighter shade (eg. instead of red, maybe #e57373 would look better). You can see all the list of colours here, I copied the hex code for `red lighten-2`.

> The problem why your solution does not work properly is that the origin states get translated while displayed in the frontend, so your row.getData().origin_state == ... will be false. Currently I don't see what would be the solution here, maybe @luksan47 has an idea.

Applying the transformations for both side (the translation and a toUpper() call) would solve the issue, since one already had it. (Not the best, I know, this part should be a bit refactored when we have the capacity.)

vo689bj committed:
Merge branch 'uran-2.x' into Use-colors-to-flag-print-tate

vo689bj committed:
Update history.blade.php

Updade color #e57373 for error instead of red and rewrite the section of code by kdmnk.

vo689bj commented:
@luksan47 I accidentally deleted the branch sorry for that i restored and reopened this. I have issues with pull request, by the way i updated your requests, can you check my update commit with color and some code changes? thank you. :)

luksan47 commented:
Please see https://github.com/luksan47/mars#important-notice
